Test/JsonDiffTest.cs

```diff
@@ -0,0 +1,120 @@
+using c_sharp_json_diff;
+using Newtonsoft.Json.Linq;
+using NUnit.Framework;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Test
+{
+    public class JsonDiffTest
+    {
+        [SetUp]
+        public void Setup()
+        {
+        }
+
+        [Test, Category("Unit")]
+        [TestCaseSource("ProviderForNumericDiffTest")]
+        public void Given_TwoJsonStrings_When_NumericDiff_Then_ExpectedResultReturned(string left, string right, string expected)
```

Comment:
Split this test into multiple functions where each function tests a single thing. That makes debugging easier.

You can use the same test case source for all of them.